golem-common/src/model/mod.rs

```diff
@@ -1031,6 +1031,62 @@ pub struct WorkerStatusRecord {
     pub total_linear_memory_size: u64,
     pub owned_resources: HashMap<WorkerResourceId, WorkerResourceDescription>,
     pub oplog_idx: OplogIndex,
+    pub active_plugins: HashSet<PluginInstallationId>,
+}
+
+impl ::bincode::Decode for WorkerStatusRecord {
+    fn decode<__D: Decoder>(decoder: &mut __D) -> Result<Self, DecodeError> {
+        Ok(Self {
+            status: Decode::decode(decoder)?,
+            deleted_regions: Decode::decode(decoder)?,
+            overridden_retry_config: Decode::decode(decoder)?,
+            pending_invocations: Decode::decode(decoder)?,
+            pending_updates: Decode::decode(decoder)?,
+            failed_updates: Decode::decode(decoder)?,
+            successful_updates: Decode::decode(decoder)?,
+            invocation_results: Decode::decode(decoder)?,
+            current_idempotency_key: Decode::decode(decoder)?,
+            component_version: Decode::decode(decoder)?,
+            component_size: Decode::decode(decoder)?,
+            total_linear_memory_size: Decode::decode(decoder)?,
+            owned_resources: Decode::decode(decoder)?,
+            oplog_idx: Decode::decode(decoder)?,
+            active_plugins: Decode::decode(decoder).or_else(|err| {
```

Comment:
This doesn't look so nice. Can we include some versioned metadata or something like this past this point so we can evolve this in the future?

I understand that you have a lib that can handle the evolution, maybe we can serialize everything past oplog_idx using your lib and put it in here.
During deserialization we can check for UnexpectedEnd like now, otherwise start reading data with your decoder including schema evolution.